unchanged:
--- chromium-38.0.2125.66.orig/breakpad/src/build/common.gypi	2014-09-18 06:53:39.000000000 +0400
+++ chromium-38.0.2125.66/breakpad/src/build/common.gypi	2014-09-28 01:52:59.870268298 +0400
@@ -725,8 +725,6 @@
             'conditions': [
               ['disable_sse2==0', {
                 'cflags': [
-                  '-march=pentium4',
-                  '-msse2',
                   '-mfpmath=sse',
                 ],
               }],
unchanged:
--- chromium-38.0.2125.66.orig/build/common.gypi	2014-09-18 06:53:33.000000000 +0400
+++ chromium-38.0.2125.66/build/common.gypi	2014-09-28 01:36:31.332624519 +0400
@@ -3653,7 +3653,6 @@
                 # http://crbug.com/313032 for an example where this has "bit"
                 # us in the past.
                 'cflags': [
-                  '-msse2',
                   '-mfpmath=sse',
                   '-mmmx',  # Allows mmintrin.h for MMX intrinsics.
                   '-m32',
unchanged:
--- chromium-38.0.2125.66.orig/media/base/BUILD.gn	2014-09-18 06:53:34.000000000 +0400
+++ chromium-38.0.2125.66/media/base/BUILD.gn	2014-09-28 02:02:04.879825799 +0400
@@ -203,7 +203,7 @@
     defines += [ "DISABLE_USER_INPUT_MONITOR" ]
   }
 
-  if (cpu_arch == "x86" || cpu_arch == "x64") {
+  if (cpu_arch == "x64") {
     sources += [ "simd/convert_yuv_to_rgb_x86.cc" ]
     deps += [
       ":media_yasm",
@@ -211,6 +211,13 @@
       ":media_sse2",
     ]
   }
+  if (cpu_arch == "x86") {
+    sources += [ "simd/convert_yuv_to_rgb_x86.cc" ]
+    deps += [
+      ":media_yasm",
+      ":media_mmx",
+    ]
+  }
 
   if (is_linux || is_win) {
     sources += [
unchanged:
--- chromium-38.0.2125.66.orig/media/media.gyp	2014-09-18 06:53:34.000000000 +0400
+++ chromium-38.0.2125.66/media/media.gyp	2014-09-28 02:04:01.020958960 +0400
@@ -976,7 +976,7 @@
             }],
           ],
         }],
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="x64"', {
           'dependencies': [
             'media_asm',
             'media_mmx',
@@ -984,6 +984,15 @@
           ],
           'sources': [
             'base/simd/convert_yuv_to_rgb_x86.cc',
+          ],
+        }],
+        ['target_arch=="ia32"', {
+          'dependencies': [
+            'media_asm',
+            'media_mmx',
+          ],
+          'sources': [
+            'base/simd/convert_yuv_to_rgb_x86.cc',
           ],
         }],
         ['OS!="linux" and OS!="win"', {
diff -u chromium-38.0.2125.66/skia/skia.gyp chromium-38.0.2125.66/skia/skia.gyp
--- chromium-38.0.2125.66/skia/skia.gyp	2014-09-28 02:04:58.931501411 +0400
+++ chromium-38.0.2125.66/skia/skia.gyp	2014-09-28 14:58:54.653142219 +0400
@@ -100,13 +100,15 @@
       'conditions': [
         [ 'os_posix == 1 and OS != "mac" and OS != "android" and \
             target_arch != "arm" and target_arch != "mipsel" and \
-            target_arch != "arm64" and target_arch != "mips64el"', {
+            target_arch != "arm64" and target_arch != "mips64el" and \
+            target_arch != "ia32"', {
           'cflags': [
             '-msse2',
           ],
         }],
         [ 'target_arch != "arm" and target_arch != "mipsel" and \
-           target_arch != "arm64" and target_arch != "mips64el"', {
+           target_arch != "arm64" and target_arch != "mips64el" and \
+           target_arch != "ia32"', {
           'sources': [
             'ext/convolver_SSE2.cc',
           ],
diff -u chromium-38.0.2125.66/skia/skia_library_opts.gyp chromium-38.0.2125.66/skia/skia_library_opts.gyp
--- chromium-38.0.2125.66/skia/skia_library_opts.gyp	2014-09-28 02:05:50.174200148 +0400
+++ chromium-38.0.2125.66/skia/skia_library_opts.gyp	2014-09-28 15:06:18.299033402 +0400
@@ -40,13 +40,15 @@
       'conditions': [
         [ 'os_posix == 1 and OS != "mac" and OS != "android" and \
            target_arch != "arm" and target_arch != "arm64" and \
-           target_arch != "mipsel" and target_arch != "mips64el"', {
+           target_arch != "mipsel" and target_arch != "mips64el" and \
+           target_arch != "ia32"', {
           'cflags': [
             '-msse2',
           ],
         }],
         [ 'target_arch != "arm" and target_arch != "mipsel" and \
-           target_arch != "arm64" and target_arch != "mips64el"', {
+           target_arch != "arm64" and target_arch != "mips64el" and \
+           target_arch != "ia32"', {
           'sources': [
             '../third_party/skia/src/opts/SkBitmapFilter_opts_SSE2.cpp',
             '../third_party/skia/src/opts/SkBitmapProcState_opts_SSE2.cpp',
@@ -58,17 +60,9 @@
             '../third_party/skia/src/opts/SkUtils_opts_SSE2.cpp',
             '../third_party/skia/src/opts/SkXfermode_opts_SSE2.cpp',
           ],
-          'conditions': [
-            [ 'target_arch != "ia32"', {
-              'dependencies': [
-                'skia_opts_ssse3',
-                'skia_opts_sse4',
-              ],
-            }, {
-              'dependencies': [
-                'skia_opts_ssse3',
-              ],
-            }],
+          'dependencies': [
+            'skia_opts_ssse3',
+            'skia_opts_sse4',
           ],
         }],
         [ 'target_arch == "arm"', {
unchanged:
--- chromium-38.0.2125.66.orig/third_party/skia/gyp/common_conditions.gypi	2014-09-18 06:55:00.000000000 +0400
+++ chromium-38.0.2125.66/third_party/skia/gyp/common_conditions.gypi	2014-09-28 01:38:39.510704411 +0400
@@ -35,7 +35,6 @@
     # As of M35, Chrome requires SSE2 on x86 (and SSSE3 on Mac).
     [ 'skia_arch_type == "x86"', {
       'cflags': [
-        '-msse2',
         '-mfpmath=sse',
       ],
     }],
unchanged:
--- chromium-38.0.2125.66.orig/third_party/skia/gyp/opts.gyp	2014-09-18 06:55:00.000000000 +0400
+++ chromium-38.0.2125.66/third_party/skia/gyp/opts.gyp	2014-09-28 01:38:24.918983219 +0400
@@ -35,13 +35,6 @@
       ],
       'conditions': [
         [ 'skia_arch_type == "x86" and skia_os != "ios"', {
-          'conditions': [
-            [ 'skia_os in ["linux", "freebsd", "openbsd", "solaris", "nacl", "chromeos", "android"]', {
-              'cflags': [
-                '-msse2',
-              ],
-            }],
-          ],
           'include_dirs': [
             '../include/utils',
           ],
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/common_audio/BUILD.gn	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/common_audio/BUILD.gn	2014-09-28 01:48:54.975441633 +0400
@@ -13,7 +13,7 @@
   # TODO(andrew): Implement.
 }
 
-if (cpu_arch == "x86" || cpu_arch == "x64") {
+if (cpu_arch == "x64") {
   source_set("common_audio_sse2") {
     # TODO(andrew): Implement.
   }
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/common_audio/common_audio.gyp	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/common_audio/common_audio.gyp	2014-09-28 01:48:06.604301846 +0400
@@ -94,7 +94,7 @@
         'vad/vad_sp.h',
       ],
       'conditions': [
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="x64"', {
           'dependencies': ['common_audio_sse2',],
         }],
         ['target_arch=="arm" or target_arch=="armv7"', {
@@ -150,7 +150,7 @@
     },
   ],  # targets
   'conditions': [
-    ['target_arch=="ia32" or target_arch=="x64"', {
+    ['target_arch=="x64"', {
       'targets': [
         {
           'target_name': 'common_audio_sse2',
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/modules/audio_processing/audio_processing.gypi	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/modules/audio_processing/audio_processing.gypi	2014-09-28 01:50:40.798955067 +0400
@@ -128,7 +128,7 @@
             'ns/windows_private.h',
           ],
         }],
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="x64"', {
           'dependencies': ['audio_processing_sse2',],
         }],
         ['(target_arch=="arm" and arm_version==7) or target_arch=="armv7"', {
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/modules/desktop_capture/BUILD.gn	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/modules/desktop_capture/BUILD.gn	2014-09-28 01:50:09.227022564 +0400
@@ -10,7 +10,7 @@
 import("../../build/webrtc.gni")
 
 use_desktop_capture_differ_sse2 =
-    (!is_ios && (cpu_arch == "x86" || cpu_arch == "x64"))
+    (!is_ios && (cpu_arch == "x64"))
 
 source_set("desktop_capture") {
   sources = [
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/modules/desktop_capture/desktop_capture.gypi	2014-09-28 01:49:41.067108879 +0400
@@ -92,7 +92,7 @@
         "x11/x_server_pixel_buffer.h",
       ],
       'conditions': [
-        ['OS!="ios" and (target_arch=="ia32" or target_arch=="x64")', {
+        ['OS!="ios" and (target_arch=="x64")', {
           'dependencies': [
             'desktop_capture_differ_sse2',
           ],
unchanged:
--- chromium-38.0.2125.66.orig/third_party/webrtc/modules/video_processing/BUILD.gn	2014-09-18 06:54:11.000000000 +0400
+++ chromium-38.0.2125.66/third_party/webrtc/modules/video_processing/BUILD.gn	2014-09-28 01:51:09.050854254 +0400
@@ -8,7 +8,7 @@
 
 import("../../build/webrtc.gni")
 
-build_video_processing_sse2 = cpu_arch == "x86" || cpu_arch == "x64"
+build_video_processing_sse2 = cpu_arch == "x64"
 
 source_set("video_processing") {
   sources = [
unchanged:
--- chromium-38.0.2125.66.orig/v8/src/ia32/assembler-ia32.cc	2014-09-18 06:54:42.000000000 +0400
+++ chromium-38.0.2125.66/v8/src/ia32/assembler-ia32.cc	2014-09-28 02:08:57.175123752 +0400
@@ -51,7 +51,6 @@
 
 void CpuFeatures::ProbeImpl(bool cross_compile) {
   base::CPU cpu;
-  CHECK(cpu.has_sse2());  // SSE2 support is mandatory.
   CHECK(cpu.has_cmov());  // CMOV support is mandatory.
 
   // Only use statically determined features for cross compile (snapshot).
unchanged:
--- chromium-38.0.2125.66.orig/third_party/qcms/BUILD.gn	2014-09-18 06:33:48.000000000 +0400
+++ chromium-38.0.2125.66/third_party/qcms/BUILD.gn	2014-09-28 02:59:13.984647815 +0400
@@ -26,8 +26,10 @@
   direct_dependent_configs = [ ":qcms_config" ]
 
   if (cpu_arch == "x86" || cpu_arch == "x64") {
-    defines = [ "SSE2_ENABLE" ]
-    sources += [ "src/transform-sse2.c" ]
+    if (cpu_arch == "x64") {
+      defines = [ "SSE2_ENABLE" ]
+      sources += [ "src/transform-sse2.c" ]
+    }
     if (!(is_win && cpu_arch == "x64")) {
       # QCMS assumes this target isn't compiled since MSVC x64 doesn't support
       # the MMX intrinsics present in the SSE1 code.
unchanged:
--- chromium-38.0.2125.66.orig/third_party/qcms/qcms.gyp	2014-09-18 06:33:48.000000000 +0400
+++ chromium-38.0.2125.66/third_party/qcms/qcms.gyp	2014-09-28 03:00:30.198377268 +0400
@@ -31,7 +31,12 @@
       'msvs_disabled_warnings': [ 4018 ],
 
       'conditions': [
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="ia32"', {
+          'sources': [
+            'src/transform-sse1.c',
+          ],
+        }],
+        ['target_arch=="x64"', {
           'defines': [
             'SSE2_ENABLE',
           ],
unchanged:
--- chromium-38.0.2125.66.orig/v8/BUILD.gn	2014-09-18 06:54:42.000000000 +0400
+++ chromium-38.0.2125.66/v8/BUILD.gn	2014-09-28 14:19:56.892099220 +0400
@@ -856,37 +856,37 @@
 
   if (v8_target_arch == "x86") {
     sources += [
-      "src/ia32/assembler-ia32-inl.h",
-      "src/ia32/assembler-ia32.cc",
-      "src/ia32/assembler-ia32.h",
-      "src/ia32/builtins-ia32.cc",
-      "src/ia32/code-stubs-ia32.cc",
-      "src/ia32/code-stubs-ia32.h",
-      "src/ia32/codegen-ia32.cc",
-      "src/ia32/codegen-ia32.h",
-      "src/ia32/cpu-ia32.cc",
-      "src/ia32/debug-ia32.cc",
-      "src/ia32/deoptimizer-ia32.cc",
-      "src/ia32/disasm-ia32.cc",
-      "src/ia32/frames-ia32.cc",
-      "src/ia32/frames-ia32.h",
-      "src/ia32/full-codegen-ia32.cc",
-      "src/ia32/ic-ia32.cc",
-      "src/ia32/lithium-codegen-ia32.cc",
-      "src/ia32/lithium-codegen-ia32.h",
-      "src/ia32/lithium-gap-resolver-ia32.cc",
-      "src/ia32/lithium-gap-resolver-ia32.h",
-      "src/ia32/lithium-ia32.cc",
-      "src/ia32/lithium-ia32.h",
-      "src/ia32/macro-assembler-ia32.cc",
-      "src/ia32/macro-assembler-ia32.h",
-      "src/ia32/regexp-macro-assembler-ia32.cc",
-      "src/ia32/regexp-macro-assembler-ia32.h",
-      "src/ia32/stub-cache-ia32.cc",
-      "src/compiler/ia32/code-generator-ia32.cc",
-      "src/compiler/ia32/instruction-codes-ia32.h",
-      "src/compiler/ia32/instruction-selector-ia32.cc",
-      "src/compiler/ia32/linkage-ia32.cc",
+      "src/x87/assembler-x87-inl.h",
+      "src/x87/assembler-x87.cc",
+      "src/x87/assembler-x87.h",
+      "src/x87/builtins-x87.cc",
+      "src/x87/code-stubs-x87.cc",
+      "src/x87/code-stubs-x87.h",
+      "src/x87/codegen-x87.cc",
+      "src/x87/codegen-x87.h",
+      "src/x87/cpu-x87.cc",
+      "src/x87/debug-x87.cc",
+      "src/x87/deoptimizer-x87.cc",
+      "src/x87/disasm-x87.cc",
+      "src/x87/frames-x87.cc",
+      "src/x87/frames-x87.h",
+      "src/x87/full-codegen-x87.cc",
+      "src/x87/ic-x87.cc",
+      "src/x87/lithium-codegen-x87.cc",
+      "src/x87/lithium-codegen-x87.h",
+      "src/x87/lithium-gap-resolver-x87.cc",
+      "src/x87/lithium-gap-resolver-x87.h",
+      "src/x87/lithium-x87.cc",
+      "src/x87/lithium-x87.h",
+      "src/x87/macro-assembler-x87.cc",
+      "src/x87/macro-assembler-x87.h",
+      "src/x87/regexp-macro-assembler-x87.cc",
+      "src/x87/regexp-macro-assembler-x87.h",
+      "src/x87/stub-cache-x87.cc",
+      "src/compiler/x87/code-generator-x87.cc",
+      "src/compiler/x87/instruction-codes-x87.h",
+      "src/compiler/x87/instruction-selector-x87.cc",
+      "src/compiler/x87/linkage-x87.cc",
     ]
   } else if (v8_target_arch == "x64") {
     sources += [
