unchanged:
--- chromium-38.0.2125.66.orig/breakpad/src/build/common.gypi	2014-09-18 06:53:39.000000000 +0400
+++ chromium-38.0.2125.66/breakpad/src/build/common.gypi	2014-09-28 01:52:59.870268298 +0400
@@ -725,8 +725,6 @@
             'conditions': [
               ['disable_sse2==0', {
                 'cflags': [
-                  '-march=pentium4',
-                  '-msse2',
                   '-mfpmath=sse',
                 ],
               }],
unchanged:
--- chromium-38.0.2125.66.orig/build/common.gypi	2014-09-18 06:53:33.000000000 +0400
+++ chromium-38.0.2125.66/build/common.gypi	2014-09-28 01:36:31.332624519 +0400
@@ -3653,7 +3653,6 @@
                 # http://crbug.com/313032 for an example where this has "bit"
                 # us in the past.
                 'cflags': [
-                  '-msse2',
                   '-mfpmath=sse',
                   '-mmmx',  # Allows mmintrin.h for MMX intrinsics.
                   '-m32',
unchanged:
diff -u chromium-38.0.2125.66/skia/skia.gyp chromium-38.0.2125.66/skia/skia.gyp
--- chromium-38.0.2125.66/skia/skia.gyp	2014-09-28 02:04:58.931501411 +0400
+++ chromium-38.0.2125.66/skia/skia.gyp	2014-09-28 14:58:54.653142219 +0400
@@ -100,13 +100,15 @@
       'conditions': [
         [ 'os_posix == 1 and OS != "mac" and OS != "android" and \
             target_arch != "arm" and target_arch != "mipsel" and \
-            target_arch != "arm64" and target_arch != "mips64el"', {
+            target_arch != "arm64" and target_arch != "mips64el" and \
+            target_arch != "ia32"', {
           'cflags': [
             '-msse2',
           ],
         }],
         [ 'target_arch != "arm" and target_arch != "mipsel" and \
-           target_arch != "arm64" and target_arch != "mips64el"', {
+           target_arch != "arm64" and target_arch != "mips64el" and \
+           target_arch != "ia32"', {
           'sources': [
             'ext/convolver_SSE2.cc',
           ],
--- chromium-38.0.2125.66.orig/third_party/skia/gyp/common_conditions.gypi	2014-09-18 06:55:00.000000000 +0400
+++ chromium-38.0.2125.66/third_party/skia/gyp/common_conditions.gypi	2014-09-28 01:38:39.510704411 +0400
@@ -35,7 +35,6 @@
     # As of M35, Chrome requires SSE2 on x86 (and SSSE3 on Mac).
     [ 'skia_arch_type == "x86"', {
       'cflags': [
-        '-msse2',
         '-mfpmath=sse',
       ],
     }],
unchanged:
--- chromium-38.0.2125.66.orig/third_party/skia/gyp/opts.gyp	2014-09-18 06:55:00.000000000 +0400
+++ chromium-38.0.2125.66/third_party/skia/gyp/opts.gyp	2014-09-28 01:38:24.918983219 +0400
@@ -35,13 +35,6 @@
       ],
       'conditions': [
         [ 'skia_arch_type == "x86" and skia_os != "ios"', {
-          'conditions': [
-            [ 'skia_os in ["linux", "freebsd", "openbsd", "solaris", "nacl", "chromeos", "android"]', {
-              'cflags': [
-                '-msse2',
-              ],
-            }],
-          ],
           'include_dirs': [
             '../include/utils',
           ],
unchanged:
--- chromium-38.0.2125.66.orig/third_party/qcms/qcms.gyp	2014-09-18 06:33:48.000000000 +0400
+++ chromium-38.0.2125.66/third_party/qcms/qcms.gyp	2014-09-28 03:00:30.198377268 +0400
@@ -31,7 +31,12 @@
       'msvs_disabled_warnings': [ 4018 ],
 
       'conditions': [
-        ['target_arch=="ia32" or target_arch=="x64"', {
+        ['target_arch=="ia32"', {
+          'sources': [
+            'src/transform-sse1.c',
+          ],
+        }],
+        ['target_arch=="x64"', {
           'defines': [
             'SSE2_ENABLE',
           ],
unchanged:
diff -Naurd chromium-38.0.2125.66.orig/v8/build/standalone.gypi chromium-38.0.2125.66/v8/build/standalone.gypi
--- chromium-38.0.2125.66.orig/v8/build/standalone.gypi	2014-09-18 06:54:42.000000000 +0400
+++ chromium-38.0.2125.66/v8/build/standalone.gypi	2014-10-02 03:07:01.107113986 +0400
@@ -95,6 +95,11 @@
     'icu_gyp_path': '../third_party/icu/icu.gyp',
 
     'conditions': [
+      ['host_arch=="ia32"', {
+        'variables': {
+          'v8_target_arch%': 'x87',
+         },
+      }],
       ['(v8_target_arch=="arm" and host_arch!="arm") or \
         (v8_target_arch=="arm64" and host_arch!="arm64") or \
         (v8_target_arch=="mipsel" and host_arch!="mipsel") or \
diff -Naurd chromium-38.0.2125.66.orig/v8/build/toolchain.gypi chromium-38.0.2125.66/v8/build/toolchain.gypi
--- chromium-38.0.2125.66.orig/v8/build/toolchain.gypi	2014-09-18 06:54:42.000000000 +0400
+++ chromium-38.0.2125.66/v8/build/toolchain.gypi	2014-10-02 03:04:11.954351938 +0400
@@ -92,6 +92,11 @@
     'android_webview_build%': '<(android_webview_build)',
   },
   'conditions': [
+    ['host_arch=="ia32"', {
+      'variables': {
+        'v8_target_arch%': 'x87',
+       },
+    }],
     ['host_arch=="ia32" or host_arch=="x64" or clang==1', {
       'variables': {
         'host_cxx_is_biarch%': 1,
diff -Naurd chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/DirectConvolver.cpp chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/DirectConvolver.cpp
--- chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/DirectConvolver.cpp	2014-09-18 07:13:23.000000000 +0400
+++ chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/DirectConvolver.cpp	2014-10-02 08:36:33.634913682 +0400
@@ -39,7 +39,7 @@
 #include "platform/audio/VectorMath.h"
 #include "wtf/CPU.h"
 
-#if (CPU(X86) || CPU(X86_64)) && !(OS(MACOSX) || USE(WEBAUDIO_IPP))
+#if (CPU(X86_64)) && !(OS(MACOSX) || USE(WEBAUDIO_IPP))
 #include <emmintrin.h>
 #endif
 
diff -Naurd chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/SincResampler.cpp chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/SincResampler.cpp
--- chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/SincResampler.cpp	2014-09-18 07:13:23.000000000 +0400
+++ chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/SincResampler.cpp	2014-10-02 08:36:55.897971773 +0400
@@ -36,7 +36,7 @@
 #include "wtf/CPU.h"
 #include "wtf/MathExtras.h"
 
-#if CPU(X86) || CPU(X86_64)
+#if CPU(X86_64)
 #include <emmintrin.h>
 #endif
 
diff -Naurd chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/VectorMath.cpp chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/VectorMath.cpp
--- chromium-38.0.2125.66.orig/third_party/WebKit/Source/platform/audio/VectorMath.cpp	2014-09-18 07:13:23.000000000 +0400
+++ chromium-38.0.2125.66/third_party/WebKit/Source/platform/audio/VectorMath.cpp	2014-10-02 08:37:10.465351519 +0400
@@ -35,7 +35,7 @@
 #include <Accelerate/Accelerate.h>
 #endif
 
-#if CPU(X86) || CPU(X86_64)
+#if CPU(X86_64)
 #include <emmintrin.h>
 #endif
 
--- chromium-38.0.2125.66/skia/ext/convolver.h.orig	2014-09-18 06:53:34.000000000 +0400
+++ chromium-38.0.2125.66/skia/ext/convolver.h	2014-10-02 18:44:47.840659298 +0400
@@ -16,7 +16,7 @@
 // We can build SSE2 optimized versions for all x86 CPUs
 // except when building for the IOS emulator.
 #if defined(ARCH_CPU_X86_FAMILY) && !defined(OS_IOS)
-#define SIMD_SSE2 1
+#define SIMD_SSE2 0
 #define SIMD_PADDING 8  // 8 * int16
 #endif
 
